system/src/Grav/Common/Assets.php

```diff
@@ -869,6 +898,49 @@ protected function pipelineJs($group = 'head', $returnURL = true)
     }
 
     /**
+     * Compresses the given file.
+     *
+     * @param string $filename  File to compress
+     * @return bool             true if succesful, false if not.
+     */
+    protected function compressFile($filename = null)
+    {
+        // Test if we have gzencode()
+        if (!function_exists('gzencode')) {
+            return false;
+        }
+
+        // Test if $filename is set
+        if (!empty($filename)) {
+
+            // Test if the source file exists and can be read
+            if (!is_readable($filename)) {
+                return false;
+            }
+
+            // Test if we can write to the target file
+            $fp = fopen($filename . ".gz", "w");
+            if (!$fp) {
+                return false;
+            }
+
+            // Load source file, gzencode it, and write it out
+            $written = fwrite($fp, gzencode(file_get_contents($filename), $this->precompress_level));
+            fclose($fp);
+
+            if ($written === false) {
+                return false;
+            }
+
+            // Make the file modification time identical to that of the source file
+            touch($filename . ".gz", filemtime($filename));
+
+            return true;
+        }
+        return false;
+    }
+
```

Comment:
is it not possible to perform this functionality with the `RocketTheme\Toolbox\File\File` class? We use this class throughout Grav to read/save files.  That way it would be more consistent an not need all the manual low-level PHP file calls.